Education/MyTask/Refrigerator.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MyTask
+{
+    public class Refrigerator
+    {
+        private DoorState UpperDoor { get; set; }
+        private DoorState LowerDoor { get; set; }
+        private PowerState FridgePowerState { get; set; }
+
+        public event RefrigeratorDelegate FrigeEvent;
+
+        public Refrigerator(DoorState upperDoor, DoorState lowerDoor, PowerState fridgePowerState)
+        {
+            this.UpperDoor = upperDoor;
+            this.LowerDoor = lowerDoor;
+            this.FridgePowerState = fridgePowerState;
+        }
+
+        public Refrigerator()
+        {
+        }
+
+        public void ChangeUpperDoorState()
+        {
+            if (UpperDoor == DoorState.Open)
+            {
+                this.UpperDoor = DoorState.Close;
+            }
+
+            else
+            {
+                this.UpperDoor = DoorState.Open;
+            }
+
+            OnChange(this, new FridgeEventArgs(this.UpperDoor, this.LowerDoor, this.FridgePowerState));
```

Comment:
This approach allows you to call event and sent correct data.
But according to the task description we should have known what exactly user did with fridge: opened door or turned it off.
In your case we obtain the info about changed state only.

Reply:
-2 to logic.